core/o11y/src/lib.rs

```diff
@@ -119,32 +179,17 @@ pub async fn default_subscriber(
         ColorOutput::Auto => std::env::var_os("NO_COLOR").is_none() && is_terminal(),
     };
 
-    let subscriber_builder = tracing_subscriber::FmtSubscriber::builder()
-        .with_ansi(ansi)
-        // Synthesizing ENTER and CLOSE events lets us log durations of spans to the log.
-        .with_span_events(
-            tracing_subscriber::fmt::format::FmtSpan::ENTER
-                | tracing_subscriber::fmt::format::FmtSpan::CLOSE,
-        )
-        .with_writer(writer)
-        .with_env_filter(log_filter)
-        .with_filter_reloading();
-    let reload_handle = subscriber_builder.reload_handle();
-    ENV_FILTER_RELOAD_HANDLE.set(reload_handle).unwrap();
+    LOG_LAYER_DATA.set((writer, ansi)).unwrap();
```

Comment:
I think needing to do something like this warrants refactoring how the reloading and log layer are handled from the principles.

Or… What I think could probably be cleaner is to modify just the filter that's in the handle using [`reload::Handle::modify`](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/reload/struct.Handle.html#method.modify)…

Reply:
`modify` makes everything much simpler. Thank you